### ISSUE TITLE
Glance card to RSTP URL drains battery even with "preload" off and card on "auto"

### ISSUE THREAD

**Issue by jasongabler** [tsightler/ring-mqtt#577]
#### Describe the problem
I have a Home Assistant generic camera entity with the stream_source set to the ring-mqtt addon's RTSP URL as found in its logs (rtsp://03cabcc9-ring-mqtt:8554/xxxxxxxxxxxx_live).  The generic camera has the "Preload camera stream" option turned off. I display the camera's stream with a glance card which has the "Camera view" set to "auto".  I click the card and the pop-up shows the stream. Works well so far.  Despite those settings, however, I believe mqtt-ring is still always preloading, always streaming.  If I have such a glance card as described above, the door bell's battery usually drains from 100% to 0% in about 8 hours.

The only variability here is that if I have no such glance cards, the battery does not drain. If I do have such a card and that card is not on a currently viewed dashboard, sometimes the battery will drain and sometimes it won't.  If I do have such a card and the card is on the currently viewed dashboard, it will always drain the battery. (Note that "currently viewed" means the card is just appearing in its default state on the dashboard. I'm not talking about clicking it to pop up the open stream.)

The currently last piece of the puzzle is trying to follow your FAQ after the part that discusses "Preload camera stream" and "Camera view" settings, where there's a suggestion to configure a password for the stream to keep potentially unknown streaming clients from continually requesting the stream and thus draining the battery.  Unfortunately, I have not been able to get the password configuration to work correctly.  When I enter values for "livestream_user" and "livestream_pass" I can no longer get the generic camera to connect to the stream. I've tried putting very simple user/pass combos into the RSTP URL and also tried using generic camera's "username" and "password" attributes. As long as ring-mqtt is configured with a user/pass, it won't work. The only way the card can access the camera stream is with no stream password at all.  So, unfortunately I cannot try this last option.

##### Describe your environment
Home Assistant 2023.3.1
Supervisor 2023.01.1
Operating System 9.5
Frontend 20230302.0 - latest
Mosquitto broker - 6.1.3

#### Describe any steps you've taken to attempt to resolve the problem
Other than working on the above configuration options,

* I've reinstalled/reconfigured from scratch a number of times.
* All the varied levels of restart/reboot after most changes
* Tried to look for telltale info in the logs, but I don't really know what I'm looking at. 

#### Debug Logs
Rather than post logs upon logs and being a passive bystander, I'm fine with being instructed on what to look for and how best to retrieve further info (including logs) for this situation.  Just tell me what you need.

**Comment by tsightler** [comment]
The logs are very clear as to when connections are made to the streaming server, the IP of the client originating the request, etc.  I would suggest just looking at the logs and seeing what is happening as I can't possibly venture to guess without them.

**Comment by jasongabler** [comment]
> 

Hmmm, I was thinking the more direct approach would be to try to get the user/pass option working as you suggest in your FAQ. Doesn't that seem like a more efficient way to tackle the situation of wanting only a single, authorized client to connect rather than tracking down and somehow blocking perhaps multiple, mystery clients?

**Comment by jasongabler** [comment]
Also, if this is a mystery client issue why does this only happen when I have a glance card? The card doesn't provide access.  So this most likely also discounts that having a user/pass could fix the problem and I'm back at square one. :p

**Comment by tsightler** [comment]
Maybe, but there's not much to putting a username/password, you set the livestream_user/password config and then add it to the RTSP URL, nothing rocket science here.  Also, the logs make it really clear what is connecting to the server by IP address anyway.  Mostly, you just have to figure it out what that is and why it is triggering.

Are you even sure it is a stream that is draining the battery?  It could be snapshots or something else.  Again, logs tell you everything that is happening, I believe starting with the only thing that actually tells you anything about what is happening is the right way to troubleshoot an issue.

**Comment by bcrowie** [comment]
I am having an issue with the battery draining as well. After the latest update my ring doorbell battery has died even while not viewing the stream.

**Comment by tsightler** [comment]
Closing this until somebody provides logs.  The logs clearly indicate everything that is happening, without logs, this is just a waste.

**Comment by jasongabler** [comment]
> Closing this until somebody provides logs. The logs clearly indicate everything that is happening, without logs, this is just a waste.

That's cool. To be upfront, I'm not going to take the time to set up debugging and all that or continue further. It's not because of ring-mqtt or you. I've been trying to get HA to do what I want with a front door camera for almost 2 years now and I'm tired of trying. I have Aarlo and Ring. I looked at ring-mqtt as a last ditch effort to have a live stream start upon a given event, where it takes no UI interaction to start the streaming. I.e. Doorbell has motion or is rung, I walk towards the door and before opening it I glance at (not touch) my automagically displaying, Fully Kiosk-running tablet and see who's at the door. I tried to work with Aarlo and Browser Mod and found I was lucky if the streaming popup popped up 1 out of 10 times. I finally gave up on that and then got wind of ring-mqtt. After a couple of months of finding time to play with it, I've just hit my input energy limit. There's zero assurance that even if I get past this particular issue I've posted about here that I'd be able to accomplish the automatic, interactionless streaming part.   I'm starting to think that this is just not possible with HA, especially considering the closest thing to tutorial on ideas such as this resort to casting a stream to a dedicated streaming device... and glance cards and other's like it within HA cannot not parade around as devices/targets.
